lisa/sut_orchestrator/azure/platform_.py

```diff
@@ -2016,8 +2050,17 @@ def _get_deployable_vhd_path(
         """
         matches = SAS_URL_PATTERN.match(vhd_path)
         if not matches:
-            return vhd_path
-        log.debug("found the vhd is a sas url, it may need to be copied.")
+            vhd_path = self._check_and_generate_sas_url(vhd_path, location, log)
+            matches = SAS_URL_PATTERN.match(vhd_path)
+            if not matches:
+                return vhd_path
+            log.debug(
```

Comment:
Why not direct copy without sas token?

Reply:
I tired, but doesn't work, the exception is 
```
azure.core.exceptions.ClientAuthenticationError: Server failed to authenticate the request. Please refer to the information in the www-authenticate header.
RequestId:ccc36585-601e-000e-2959-a6a9db000000
Time:2022-08-02T10:17:54.9250330Z
ErrorCode:CannotVerifyCopySource
Content: <?xml version="1.0" encoding="utf-8"?><Error><Code>CannotVerifyCopySource</Code><Message>Server failed to authenticate the request. Please refer to the information in the www-authenticate header.
RequestId:ccc36585-601e-000e-2959-a6a9db000000
Time:2022-08-02T10:17:54.9250330Z</Message></Error>
```

Reply:
probably I need find out another way, for this method start_copy_from_url, it needs to be ` The source blob must either be public or must be authenticated via a shared access signature. `

![image](https://user-images.githubusercontent.com/10083705/182559339-94348bc9-e7a9-4572-8a5b-65611f47cfc4.png)